codec/src/main/java/io/netty/handler/codec/compression/LzfEncoder.java

```diff
@@ -17,27 +17,42 @@
 
 import com.ning.compress.BufferRecycler;
 import com.ning.compress.lzf.ChunkEncoder;
+import com.ning.compress.lzf.LZFChunk;
 import com.ning.compress.lzf.LZFEncoder;
 import com.ning.compress.lzf.util.ChunkEncoderFactory;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.MessageToByteEncoder;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
 
-import static com.ning.compress.lzf.LZFChunk.*;
+import static com.ning.compress.lzf.LZFChunk.MAX_CHUNK_LEN;
 
 /**
  * Compresses a {@link ByteBuf} using the LZF format.
- *
+ * <p>
  * See original <a href="http://oldhome.schmorp.de/marc/liblzf.html">LZF package</a>
  * and <a href="https://github.com/ning/compress/wiki/LZFFormat">LZF format</a> for full description.
  */
 public class LzfEncoder extends MessageToByteEncoder<ByteBuf> {
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(LzfEncoder.class);
+
     /**
      * Minimum block size ready for compression. Blocks with length
      * less than {@link #MIN_BLOCK_TO_COMPRESS} will write as uncompressed.
      */
     private static final int MIN_BLOCK_TO_COMPRESS = 16;
 
+    /**
+     * Compress threshold for LZF format. When the amount of input data is less than compressThreshold,
+     * we will construct an uncompressed output according to the LZF format.
+     * <p>
+     * When the value is less than {@see ChunkEncoder#MIN_BLOCK_TO_COMPRESS}, since LZF will not compress data
+     * that is less than {@see ChunkEncoder#MIN_BLOCK_TO_COMPRESS}, compressThreshold will not work.
```

Comment:
should we validate this on construction and if so throw ?

Reply:
Currently I checked `compressThreshold <MIN_BLOCK_TO_COMPRESS`. If so, `compressThreshold = MIN_BLOCK_TO_COMPRESS`, and print a log.

It seems that the startup phase informs the user that this is an inappropriate value and is a good choice. fixed...